src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -152,6 +155,14 @@ public void launch() {
         // referenced by main/dependencies of main
         final Queue<String> autostart = findBuiltInServicesAndPlugins(); //NOPMD
         loadPlugins();
+
+        // Start MqttProxyIPCService after plugins are loaded, as it requires
+        // DiskSpooler Implementation Plugin. This behavior is only needed in testing
+        // as we scan our own classpath to find the @ImplementsService
```

Comment:
Do you know if we have any other plugin components that may require publishing to iot core? If any, would they break since now they are loaded before MqttProxyIPCService?

Reply:
Not that I know of. I believe none of them would do any publishing before their startup phase and we do a startupAllServices at a later point.